safesql/src/main/java/org/dhatim/safesql/SafeSqlAppendable.java

```diff
@@ -0,0 +1,70 @@
+package org.dhatim.safesql;
+
+import java.util.stream.Stream;
+
+public interface SafeSqlAppendable {
+
+    SafeSqlAppendable param(int num);
+    SafeSqlAppendable param(long num);
+    SafeSqlAppendable param(double num);
+    SafeSqlAppendable param(boolean bool);
+    SafeSqlAppendable param(Object obj);
+    SafeSqlAppendable params(Object param1, Object param2);
+    SafeSqlAppendable params(Object param1, Object param2, Object param3);
+    SafeSqlAppendable params(Object param1, Object param2, Object param3, Object... others);
+    SafeSqlAppendable params(Object... parameters);
```

Comment:
I agree with Raphaël: we should favor readability over imaginary performance tweaks. Please do not loose time trying to optimize something you have not profiled. I bet the SQL query execution is orders of magnitude slower than its construction, anyway.